### ISSUE TITLE
Capturing request exception stacks?

### ISSUE THREAD

**Issue by ryanrupp** [stagemonitor/stagemonitor#112]
I was wondering how exception stacks get captured on request monitoring as I noticed this wasn't getting picked up on my requests. Looking into this it looks like Stagemonitor looks for an "exception" attribute on the request here - https://github.com/stagemonitor/stagemonitor/blob/master/stagemonitor-web/src/main/java/org/stagemonitor/web/monitor/MonitoredHttpRequest.java#L219 - what is supposed to be setting this though, Spring? I'm using Wildfly 9 (Undertow servlet container) + Spring 4.1.5 for reference. Otherwise if it's just something projects need to manually set so stagemonitor can pick this up I'll just add a Spring HandlerInterceptor that populates the "exception" request attribute for handler calls that ended in an exception. Thanks!

**Comment by felixbarny** [comment]
Stagemonitor also checks if the call to `filterChain.doFilter()` yields in a exception. But if the exception is handled at the servlet level (like many frameworks, including Spring do) this is not sufficient. That's why the application can set the `"exception"` request attribute. Springs built in `SimpleMappingExceptionResolver` does set this attribute. So yes, your approach would most likely work.

Maybe I should also check for a attribute with the name `javax.servlet.error.exception`. Could you debug https://github.com/stagemonitor/stagemonitor/blob/master/stagemonitor-web/src/main/java/org/stagemonitor/web/monitor/MonitoredHttpRequest.java#L219 and check if there is such an attribute available? If it is, I can add that to the code - this should be more portable if it's working.

**Comment by ryanrupp** [comment]
This is for a REST API so the way I'm sending back exceptions is Spring is something:

```
@ExceptionHandler
    @ResponseBody ErrorResponse handleAllExceptions(Exception ex, HttpServletResponse response) throws IOException {
        // code that builds out the ErrorResponse
    }
```

It looks like because of this Spring sends me down this code path - https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/DispatcherServlet.java#L1190 - so for some reason in this case Spring ends up putting the exception on the request as the attribute "org.springframework.web.servlet.DispatcherServlet.EXCEPTION"

I tried playing around with things a bit to follow the whole return a ModelAndView object which then Spring does set "javax.servlet.error.exception" here - https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/DispatcherServlet.java#L1200 - however then for some reason they clear those error attributes here - https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/DispatcherServlet.java#L1029 - before the Stagemonitor filter has control again to try reading them.

So for my case at least it worked for me just looking for "org.springframework.web.servlet.DispatcherServlet.EXCEPTION", but I'm guessing you don't want to hardcode that in Stagemonitor code. I'm not sure why Spring does this inconsistently or why they get cleared, it does seem like if it was consistent/not cleared then it would make sense to look for "javax.servlet.error.exception" too. Are you open to making it a pluggable configuration e.g. I could then have:

```
stagemonitor.requests.error-attributes=javax.servlet.error.exception,exception,org.springframework.web.servlet.DispatcherServlet.EXCEPTION
```
and stagemonitor just checks those in order using the first one that is non null.

**Comment by felixbarny** [comment]
A config seems reasonable

**Comment by ryanrupp** [comment]
Resolved in #116
